lib/setopt.c

```diff
@@ -63,6 +64,12 @@ CURLcode Curl_setstropt(char **charp, const char *s)
 
     if(str) {
       size_t len = strlen(str);
+
+      /* Protect against malicious usage of BLOB magic */
+      if((len == CURL_BLOB_MAGIC_STRLEN) &&
+          (!memcmp(s, CURL_BLOB_MAGIC, CURL_BLOB_MAGIC_SIZE)))
+        return CURLE_BAD_FUNCTION_ARGUMENT;
```

Comment:
This code is unnecessary. We *know* when there's a blob passed on since it is now a dedicated option. It can't be "malicious".